### PR TITLE
Makefile: Next version update to 24.11

### PR DESCRIPTION
**Pull request by ShahanaFarooqui**
Changelog-None.

### CODE REVIEW

**Comment by niftynei** [comment]
Seems like there's an unrelated test flake

```
**BROKEN** lightningd: DEPRECATED API USED rest-port.clnrest-prefix 
lightningd-1 2024-09-06T17:34:34.298Z UNUSUAL lightningd: Option rest-port=36301 deprecated in v23.11, renaming to clnrest-port=36301
lightningd-1 2024-09-06T17:34:34.298Z **BROKEN** lightningd: DEPRECATED API USED rest-host.clnrest-prefix 
lightningd-1 2024-09-06T17:34:34.298Z UNUSUAL lightningd: Option rest-host=127.0.0.1 deprecated in v23.11, renaming to clnrest-host=127.0.0.1
```

**Comment by ShahanaFarooqui** [comment]
@niftynei CI is fixed after adjusting some tests due to updated next cln release version.  Please review and ACK now.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (vincenzopalazzo) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: This looks like a change that required a separate commit

File: doc/contribute-to-core-lightning/release-checklist.md
Comment (vincenzopalazzo) [inline doc/contribute-to-core-lightning/release-checklist.md:1]: same for

File: tests/test_gossip.py
Comment (vincenzopalazzo) [inline tests/test_gossip.py:1]: removing logs to fix the CI sounds like a dangerous thing, probably the right fix is https://github.com/ElementsProject/lightning/pull/7663

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ShahanaFarooqui) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: This template will also require to be updated after every release cycle with the next release dates (first freeze, RC1 etc). So I wanted to bundle all changes related to post release cycle in a single PR.

File: doc/contribute-to-core-lightning/release-checklist.md
Comment (ShahanaFarooqui) [reply doc/contribute-to-core-lightning/release-checklist.md:1]: This also is the part of "post release wrap-up and prepare for next release" cycle like above. So keeping them all together.

File: tests/test_gossip.py
Comment (ShahanaFarooqui) [reply tests/test_gossip.py:1]: Adjusted other failing tests to make them compatible with the next release but this test is still testing for `listchannels()['channels']]` which has been moved to `listpeerchannels` in v23.02.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (vincenzopalazzo) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Right, but in the same PR you can create an additional commit just for this github docs updated

**Comment by rustyrussell** [comment]
I reversed the order of commits to put CI changes *first*.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ShahanaFarooqui) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Thanks, Split the commit into three separate commits now.

**Pull request merged**